Pipelines/Templates/deploy-Solution.yml

```diff
@@ -121,87 +121,9 @@ steps:
-          $pullrequestqueryResponse = Invoke-RestMethod $pullrequestqueryBodyResourceUrl -Method 'POST' -Headers $headers -Body $pullrequestqueryBody 
-          $pullrequestqueryResponseResults = $pullrequestqueryResponse.results
-          $pullRequestId = $pullrequestqueryResponseResults.$buildSourceVersion.pullRequestId
-          if (-not [string]::IsNullOrEmpty($pullRequestId)) {
-            $pullRequestLabelQuery = "$(System.TeamFoundationCollectionUri)$(System.TeamProjectId)/_apis/git/repositories/$(Build.Repository.Name)/pullRequests/$pullRequestId/labels?api-version=6.0"
-            
-            $pullRequestLabelQueryResponse = Invoke-RestMethod -Uri $pullRequestLabelQuery -Method Get -Headers @{
-              Authorization = "Bearer $env:SYSTEM_ACCESSTOKEN"
-            }
-
-            $pullRequestLabelQueryResponseValue = $pullRequestLabelQueryResponse.value
-
-            if ($pullRequestLabelQueryResponseValue.Count -gt 0) {
-              $triggerSolutionUpgrade = $pullRequestLabelQueryResponseValue.name.Contains($solutionUpgradeLabel).ToString().ToLower()
-              echo "##vso[task.setvariable variable=TriggerSolutionUpgrade;isOutput=true]$triggerSolutionUpgrade"
-            }
-          }
-        }
-        
-        if("$(Build.Repository.Provider)" -eq "GitHub") {
-          $headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
-          $headers.Add("Accept", "application/vnd.github.groot-preview+json")
-          $headers.Add("Authorization", "Bearer $(GitHubPAT)")
-          
-          Write-Host "Build.SourceVersion is $(Build.SourceVersion)"
-
-          $pullsRequest = "https://api.github.com/repos/$(GitHubRepo)/commits/$(Build.SourceVersion)/pulls"
-          
-          $pullsResponse = Invoke-RestMethod $pullsRequest -Method 'GET' -Headers $headers
-          
-          if ($pullsResponse.Count -gt 0) {
-            $pullNumber = $pullsResponse[0].number
-            $issuesRequest = "https://api.github.com/repos/$(GitHubRepo)/issues/$pullNumber"
-            $issuesResponse = Invoke-RestMethod $issuesRequest -Method 'GET' -Headers $headers 
-            
-            if ($issuesResponse.labels.Count -gt 0) {
-              $triggerSolutionUpgrade = $issuesResponse.labels.name.Contains($solutionUpgradeLabel).ToString().ToLower()
-              echo "##vso[task.setvariable variable=TriggerSolutionUpgrade;isOutput=true]$triggerSolutionUpgrade"
-            }
-          }
-        }
-      }
-    }
+    . "$env:POWERSHELLPATH/set-trigger-solution-upgrade.ps1"
+    Set-TriggerSolutionUpgrade '$(TriggerSolutionUpgrade)' '$(System.TeamFoundationCollectionUri)' '$(System.TeamProjectId)' '$(Build.Repository.Name)' '$(Build.Repository.Provider)' '$(Build.SourceVersion)' '$(Build.Reason)' '$(System.AccessToken)' $testConfig.gitHubRepo $testConfig.gitHubPat
```

Comment:
I think we need to still pass as an env: parameter ($(System.AccessToken) to this script due to the way that Azure Devops handles secret variables. This may run locally but in Azure DevOps secrets are handled differently. Should validate that a secret passed from a YAML pipeline in this manner will resolve correctly.

Reply:
What we've noticed is that variables passed from YAML to PS is that env: variables are treated differently than variables passed directly to the PS scripts. Assumption is these variables are held in memory because they may be sensitive as opposed to other variables that get added to the environment variables and are accessible from anywhere.

Reply:
I think there's an issue outstanding with a client related to this. I need to follow up on this. Tagging myself @mikefactorial

Reply:
@mikefactorial Updated script to use "env: parameter ($(System.AccessToken)" in both YAML and ps1 files.